CRM/Contribute/Tokens.php

```diff
@@ -73,28 +67,58 @@ protected function getExposedFields(): array {
       'thankyou_date',
       'tax_amount',
       'contribution_status_id',
+      'contribution_status_id:name',
+      'contribution_status_id:label',
       'financial_type_id',
+      'financial_type_id:name',
+      'financial_type_id:label',
       'payment_instrument_id',
+      'payment_instrument_id:name',
+      'payment_instrument_id:label',
+      'cancel_reason',
+      'amount_level',
+      'check_number',
     ];
+    if (CRM_Campaign_BAO_Campaign::isCampaignEnable()) {
+      $fields[] = 'campaign_id';
+      $fields[] = 'campaign_id.name';
+      $fields[] = 'campaign_id.title';
+    }
+
+    return $fields;
   }
 
-  /**
-   * Get tokens supporting the syntax we are migrating to.
-   *
-   * In general these are tokens that were not previously supported
-   * so we can add them in the preferred way or that we have
-   * undertaken some, as yet to be written, db update.
-   *
-   * See https://lab.civicrm.org/dev/core/-/issues/2650
-   *
-   * @return string[]
-   */
-  public function getBasicTokens(): array {
-    $return = [];
-    foreach ($this->getExposedFields() as $fieldName) {
-      $return[$fieldName] = $this->getFieldMetadata()[$fieldName]['title'];
+  public function getAliasTokens(): array {
+    $aliases = [];
+    if (CRM_Campaign_BAO_Campaign::isCampaignEnable()) {
+      // Unit-tests are written to use these funny tokens - but they're not valid in APIv4.
```

Comment:
the unit tests from one of the 2 PRs used them..... the other one used `campaign_id.name` - we just need to agree which syntax it is